### PR TITLE
PAAS-1245 make people aware that old resources need to be taken down …

### PR DESCRIPTION
**Pull request by grosser**
…when changing names

(since automatic deletion/re-create is super tricky and dangerous)

@zendesk/compute 

screen shot 2018-08-20 at 3 09 45 pm

### CODE REVIEW

File: plugins/kubernetes/app/views/kubernetes/roles/_form.html.erb [hunk 1]
```diff
@@ -7,9 +7,15 @@
...
     <%= form.input :service_name %>
     <%= form.input :resource_name,
-          pattern: Kubernetes::RoleValidator::VALID_LABEL,
-          help: "Name for the resource this role will create (service/job), a-Z 0-9 and -"
+        pattern: Kubernetes::RoleValidator::VALID_LABEL,
+        help: "Name for the resource this role will create (service/job), a-Z 0-9 and -"
     %>
+    <% if @role.manual_deletion_required? %>
+      <div class="alert-danger">
+        <%= form.input :manual_deletion_acknowledged, as: :check_box, label: "I will manually delete the changed resource/service from all deploy groups before deploying" %>
```
Comment (adammw) [inline plugins/kubernetes/app/views/kubernetes/roles/_form.html.erb]: before? wouldn't that cause an outage? or will renames always cause an outage?

File: plugins/kubernetes/test/models/kubernetes/role_test.rb [hunk 2]
```diff
@@ -88,6 +88,24 @@ def commit
...
+      it "is invalid when not acknowledged" do
+        refute_valid role
+      end
+
+      it "is valid when acknowledged" do
+        role.manual_deletion_acknowledged = true
+        assert_valid role
+      end
+
+      it "is valid when acknowledged" do
```
Comment (adammw) [inline plugins/kubernetes/test/models/kubernetes/role_test.rb]: dupe

**Comment by grosser** [comment]
changing service names will always cause an outage
changing deployment names without removing the old first will create 2
deployments with the same matchlabels which is kinda scary too (and
whatever other resources do like statefulset/job/pod) ...
so unless we make sure that is ok I'd go the safe route and delete first

File: plugins/kubernetes/app/views/kubernetes/roles/_form.html.erb (hunk 1 above)
Comment (adammw) [reply plugins/kubernetes/app/views/kubernetes/roles/_form.html.erb]: in that case.. maybe mention the word "outage" / "undefined behaviour" in the check box label...

**Pull request merged**